java/arcs/core/storage/DriverFactory.kt

```diff
@@ -84,6 +79,22 @@ object DriverFactory {
 
     /** Reset the driver registration to an empty set. For use in tests only. */
     fun clearRegistrations() = providers.lazySet(setOf())
+
+    /**
+     * A helper to launch a particle task on all registered providers.
+     *
+     * A supervisorScope is created by this method, so if one of the tasks fails,
+     * the failure will not cause the caller or any other tasks to fail.
+     */
+    private suspend fun runSafelyOnAllProviders(
+        block: suspend (DriverProvider) -> Unit
+    ) = supervisorScope {
+        providers.value.forEach {
+            launch {
+                block(it)
+            }
```

Comment:
It might be worth using `async` here, to collect the deferreds.. then in a loop - you could call `await` on them and collect any exceptions the child jobs threw. The question is what to do with them after that point :)

As we discussed offline, it'd probably be a good idea to aggregate them here, then re-throw a new exception from the supervisorScope so we still cancel everything up to the *next* supervisorScope, letting *it* know that something went wrong here.